File: Irony.Samples/SourceSamples/99 bottles.py
#!/usr/bin/env python
# -*- coding: iso-8859-1 -*-
"""
99 Bottles of Beer (by Gerold Penz)
Python can be simple, too :-)
"""

for quant in range(99, 0, -1):
   if quant > 1:
      print quant, "bottles of beer on the wall,", quant, "bottles of beer."
      if quant > 2:
         suffix = str(quant - 1) + " bottles of beer on the wall."
      else:
         suffix = "1 bottle of beer on the wall."
   elif quant == 1:
      print "1 bottle of beer on the wall, 1 bottle of beer."
      suffix = "no more beer on the wall!"
   print "Take one down, pass it around,", suffix
   print "--"

File: Irony.Samples/SourceSamples/99 bottles.rb
#!/usr/bin/env ruby
# 99 bottles problem in Ruby

class Wall

  def initialize(num_of_bottles)
    @bottles = num_of_bottles
  end

  def sing_1_verse
    @output = sing_num(@bottles) + " on the wall, " + sing_num(@bottles) + "\n"
    @output += "take one down, pass it around, " + sing_num(@bottles-1) + "\n\n"
    return @output
  end

  def sing_all
    @output = ''
    while @bottles > 0 do
      @output += sing_1_verse()
      @bottles -= 1
    end
    return @output
  end
  
  def sing_num(num)
    @counter = (num > 1) ? 'bottles' : 'bottle'
    "#{num} #{@counter} of beer"
  end

end # class Wall

wall = Wall.new(99)
puts wall.sing_all()
